# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/webscheduling/WebTaskScheduler.h
# Commit: f975d7f9d4b7
# Full Hash: f975d7f9d4b7b71e06bc5f9b21266d7bf41e4dad
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-04-16 04:08:18
# Description:
#   Bug 1960501 - Do not try to run delayed web tasks when the scheduler is disconnected r=mccr8
#   
#   I think this bug is about a case where the scheduler was CC'd,
#   and then the delayed tasks tried to run.
#   
# ==============================================================================

diff -r a0ce31097c15 -r f975d7f9d4b7 dom/webscheduling/WebTaskScheduler.h
--- a/dom/webscheduling/WebTaskScheduler.h	Tue Apr 15 20:32:55 2025 +0000
+++ b/dom/webscheduling/WebTaskScheduler.h	Tue Apr 15 20:39:27 2025 +0000
@@ -345,7 +345,7 @@
   NS_DECL_CYCLE_COLLECTION_CLASS(DelayedWebTaskHandler)
 
   MOZ_CAN_RUN_SCRIPT bool Call(const char* /* unused */) override {
-    if (mScheduler && mWebTask) {
+    if (mScheduler && mWebTask && mWebTask->isInList()) {
       MOZ_ASSERT(!mWebTask->HasScheduled());
       if (!mScheduler->DispatchTask(mWebTask, mPriority)) {
         return false;
